### DISCUSSION TITLE
Field 'generated_conversions' doesn't have a default value

### DISCUSSION THREAD

**Discussion by LarryBarker** [spatie/laravel-medialibrary#2675] [Q&A]
We had been using v7 of the package due to environment constraints. We recently upgraded our environment to PHP 8 and have decided to upgrade to v9 of the package.

I followed the upgrade guide as provided, and have been able to get the package working locally. However, on our QA environment, we are having issues uploading media.

When we upload media, we see the error: `Field 'generated_conversions' doesn't have a default value`

I'm trying to understand why the package would work for me locally, but not in our QA environment. Here is the migration we used to add the `generated_conversions` column:

```
if (! Schema::hasColumn('media', 'generated_conversions')) {
            Schema::table('media', function (Blueprint $table) {
                $table->json('generated_conversions')->after('custom_properties');
            });
        }
```

Here is a screen shot of the `media` table structure for my local environment:
![image](https://user-images.githubusercontent.com/28734844/142239309-330c9487-7950-4a34-a954-518f9380005a.png)

Here is a screen shot of the `media` table structure for our QA environment:
![image](https://user-images.githubusercontent.com/28734844/142239389-4fe71be1-ac5e-423e-b56f-4b93d622f6dd.png)

We also noticed in the stack trace, the `generated_conversions` column is not listed in the insert (but the value is):
```
insert into `media` (`name`, `file_name`, `disk`, `collection_name`, `mime_type`, `size`, `custom_properties`, `responsive_images`, `manipulations`, `model_id`, `model_type`, `order_column`, `updated_at`, `created_at`) values (algorithm-microservice-architecture, algorithm-microservice-architecture.pdf, gcs, resources, application/pdf, 53585, [], [], [], 22, App\\Resource, 5, 2021-11-17 18:53:37, 2021-11-17 18:53:37)
```

Any feedback is greatly appreciated. Thank you :)

**Comment by PaolaRuby** [comment]
```php
if (! Schema::hasColumn('media', 'generated_conversions')) {
    Schema::table('media', function (Blueprint $table) {
        $table->json('generated_conversions')->after('custom_properties')->nullable()->default('[]');
    });
}
```
